projects/backend/src/plugins/oauth.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const oauthPlugin = require('@fastify/oauth2')
+
+module.exports = async function spotifyOauthPlugin (app, opts, next) {
+  app.register(oauthPlugin, {
+    name: 'spotify',
+
+    scope: ['user-read-email', 'user-top-read'],
+
+    credentials: {
+      client: {
+        id: app.appConfig.SPOTIFY_CLIENT_ID,
+        secret: app.appConfig.SPOTIFY_CLIENT_SECRET
+      },
+
+      auth: oauthPlugin.SPOTIFY_CONFIGURATION
+    },
+
+    startRedirectPath: '/login/spotify',
+    callbackUri: app.appConfig.OAUTH_REDIRECT_URI
+  })
+
+  app.register(require('@fastify/jwt'), {
+    secret: 'secret'
+  })
+
+  app.decorate('authenticate', async function (request, reply) {
+    try {
+      await request.jwtVerify()
+    } catch (err) {
+      reply.send(err)
+    }
```

Comment:
when we use async, we should avoid `reply.send`

in this case, if the auth throw an error, the `errorHandler` will take care of it

```suggestion
    await request.jwtVerify()
```